### PR TITLE
to trigger assets from publish.yml only for tag builds

### PR DESCRIPTION
**Pull request by yash-zededa**
In the current setup the assets.yml is triggered when the publish.yml build is completed but the trigger is not working as expected and to release the correct tag we need to re-run the build. This PR will fix this workflow and the assets.yml will be triggered by publish.yml and the tag built will be passed to assets.yml to publish the artefacts.

### CODE REVIEW

File: .github/workflows/assets.yml
Comment (yash-zededa) [reply .github/workflows/assets.yml]: This happens when you see a lot of `${{ }}` and start replacing - this is fixed

File: .github/workflows/publish.yml
Comment (uncleDecart) [inline .github/workflows/publish.yml:159]: Why do we have restriction on repository full name here?

Comment (eriknordmark) [reply .github/workflows/publish.yml:159]: Do avoid triggering for a fork of the repo, I assume.

File: .github/workflows/assets.yml
Comment (eriknordmark) [inline .github/workflows/assets.yml:18]: Can you fix the blanks at the end of the line?

File: .github/workflows/publish.yml
Comment (eriknordmark) [inline .github/workflows/publish.yml:160]: Don't we need all of eve, verification, and manifest before we run this?

**Review by eriknordmark — APPROVED** [review]
LGTM - run tests but don't merge to make sure it doesn't interfere with cutting a 9.4.X patch release tomorrow.

File: .github/workflows/publish.yml
Comment (uncleDecart) [reply .github/workflows/publish.yml:159]: But how would you test that workflow is working correctly before merging? My approach to it would be running workflow in your own fork

**Comment by eriknordmark** [comment]
@yash-zededa should we go ahead and merge this now?

**Comment by yash-zededa** [comment]
> @yash-zededa should we go ahead and merge this now?

Apologies for the delayed response. Yes, we can go ahead and merge this

**Pull request merged**
